conversation_reconstruction_dataflow/conversation_reconstruction/construct_utils/conversation_constructor.py

```diff
@@ -16,7 +15,9 @@
 -------------------------------------------------------------------------------
```

Comment:
This file could do with more doc-strings to explain the different functions; and being broken up into simpler parts, and having tests for the parts. It's the key complex logic. :)

Reply:
Put in an issue, it takes some time to document this but will do this soon.

Reply:
Updated documentation for this gigantic code.